### PR TITLE
Lose and restart

### PR DESCRIPTION
**Pull request by NiceGuyAndersson**
The game can now be reset to a saved state.
Currently, after 10 seconds the player dies (just a hard reset)

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L651](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/bb8ed37ae83d6b37e29b656de3d82833d1f88386/src/Engine/Manager/ScriptManager.cpp#L651): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L655](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/bb8ed37ae83d6b37e29b656de3d82833d1f88386/src/Engine/Manager/ScriptManager.cpp#L655): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L659](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/bb8ed37ae83d6b37e29b656de3d82833d1f88386/src/Engine/Manager/ScriptManager.cpp#L659): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L184](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/bb8ed37ae83d6b37e29b656de3d82833d1f88386/src/Engine/Hymn.cpp#L184): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L673](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/bb8ed37ae83d6b37e29b656de3d82833d1f88386/src/Engine/Manager/ScriptManager.cpp#L673): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L681](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/bb8ed37ae83d6b37e29b656de3d82833d1f88386/src/Engine/Manager/ScriptManager.cpp#L681): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Game/Resources/Constant Scene/Minecart/Script/Player.as [hunk 1]
```diff
@@ -24,5 +25,9 @@ class Player{
         // Only control camera with mouse if we're not running in VR.
         if (!IsVRActive())
             MouseUpdate();
+            
+		deathTimer += i;
```
Comment (Chainsawkitten) [inline Game/Resources/Constant Scene/Minecart/Script/Player.as]: Horizontal tab

**Comment by Chainsawkitten** [comment]
Very spooky pull request.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L654](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf5892169316d1a35000810255670fa3579c8ff4/src/Engine/Manager/ScriptManager.cpp#L654): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L658](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf5892169316d1a35000810255670fa3579c8ff4/src/Engine/Manager/ScriptManager.cpp#L658): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L662](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf5892169316d1a35000810255670fa3579c8ff4/src/Engine/Manager/ScriptManager.cpp#L662): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L184](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf5892169316d1a35000810255670fa3579c8ff4/src/Engine/Hymn.cpp#L184): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf5892169316d1a35000810255670fa3579c8ff4/src/Engine/Manager/ScriptManager.cpp#L676): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L684](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/cf5892169316d1a35000810255670fa3579c8ff4/src/Engine/Manager/ScriptManager.cpp#L684): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**